pkg/generated/restapi/configure_rekor_server.go

```diff
@@ -174,21 +178,81 @@ func setupMiddlewares(handler http.Handler) http.Handler {
 	return handler
 }
 
+type httpRequestFields struct {
+	requestMethod string
+	requestURL    string
+	requestSize   string
+	status        int
+	responseSize  string
+	userAgent     string
+	remoteIP      string
+	latency       string
+	protocol      string
+}
+
+func (h httpRequestFields) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddString("requestMethod", h.requestMethod)
+	enc.AddString("requestUrl", h.requestURL)
+	enc.AddString("requestSize", h.requestSize)
+	enc.AddInt("status", h.status)
+	enc.AddString("responseSize", h.responseSize)
+	enc.AddString("userAgent", h.userAgent)
+	enc.AddString("remoteIP", h.remoteIP)
+	enc.AddString("latency", h.latency)
+	enc.AddString("protocol", h.protocol)
+	return nil
+}
+
 // We need this type to act as an adapter between zap and the middleware request logger.
-type logAdapter struct {
+type zapLogEntry struct {
+	r *http.Request
 }
 
-func (l *logAdapter) Print(v ...interface{}) {
-	log.Logger.Info(v...)
+func (z *zapLogEntry) Write(status, bytes int, header http.Header, elapsed time.Duration, extra interface{}) {
+	var fields []interface{}
+
+	// follows https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry as a convention
+	// append HTTP Request / Response Information
+	scheme := "http"
+	if z.r.TLS != nil {
+		scheme = "https"
+	}
+	httpRequestObj := httpRequestFields{
+		requestMethod: z.r.Method,
+		requestURL:    fmt.Sprintf("%s://%s%s", scheme, z.r.Host, z.r.RequestURI),
+		requestSize:   fmt.Sprintf("%d", z.r.ContentLength),
+		status:        status,
+		responseSize:  fmt.Sprintf("%d", bytes),
+		userAgent:     z.r.Header.Get("User-Agent"),
+		remoteIP:      z.r.RemoteAddr,
+		latency:       fmt.Sprintf("%.9fs", elapsed.Seconds()),
+		protocol:      z.r.Proto,
+	}
+	fields = append(fields, zap.Object("httpRequest", httpRequestObj))
+	if extra != nil {
+		fields = append(fields, zap.Any("extra", extra))
+	}
+
+	log.ContextLogger(z.r.Context()).With(fields...).Info("completed request")
```

Comment:
If the `fields` were declared as `[]zapcore.Field` then this can be
`log.ContextLogger(z.r.Context()).With(fields).Info("completed request")`

Reply:
`With([]interface{})` is how this is defined, and the logger doesn't work if you more strongly type this.